metricflow-semantics/metricflow_semantics/test_helpers/snapshot_helpers.py

```diff
@@ -259,41 +259,43 @@ def assert_linkable_element_set_snapshot_equal(  # noqa: D103
-                    sorted(
-                        linkable_element_property.name for linkable_element_property in linkable_dimension.properties
-                    ),
-                )
+            row_to_add = (
+                # Checking a limited set of fields as the result is large due to the paths in the object.
+                (linkable_dimension.join_path.left_semantic_model_reference.semantic_model_name,)
+                + tuple(
+                    path_element.semantic_model_reference.semantic_model_name
+                    for path_element in linkable_dimension.join_path.path_elements
+                ),
+                tuple(entity_link.element_name for entity_link in linkable_dimension.entity_links),
+                linkable_dimension.element_name,
+                linkable_dimension.time_granularity.name if linkable_dimension.time_granularity is not None else "",
+                linkable_dimension.date_part.name if linkable_dimension.date_part is not None else "",
+                sorted(linkable_element_property.name for linkable_element_property in linkable_dimension.properties),
             )
+            if row_to_add not in rows:
+                rows.append(row_to_add)
 
     for linkable_entity_iterable in linkable_element_set.path_key_to_linkable_entities.values():
         for linkable_entity in linkable_entity_iterable:
-            rows.append(
-                (
-                    # Checking a limited set of fields as the result is large due to the paths in the object.
-                    linkable_entity.semantic_model_origin.semantic_model_name,
-                    tuple(entity_link.element_name for entity_link in linkable_entity.entity_links),
-                    linkable_entity.element_name,
-                    "",
-                    "",
-                    sorted(linkable_element_property.name for linkable_element_property in linkable_entity.properties),
-                )
+            row_to_add = (
+                # Checking a limited set of fields as the result is large due to the paths in the object.
+                (linkable_entity.join_path.left_semantic_model_reference.semantic_model_name,)
+                + tuple(
+                    path_element.semantic_model_reference.semantic_model_name
+                    for path_element in linkable_entity.join_path.path_elements
+                ),
+                tuple(entity_link.element_name for entity_link in linkable_entity.entity_links),
+                linkable_entity.element_name,
+                "",
+                "",
+                sorted(linkable_element_property.name for linkable_element_property in linkable_entity.properties),
             )
+            if row_to_add not in rows:
+                rows.append(row_to_add)
 
     for linkable_metric_iterable in linkable_element_set.path_key_to_linkable_metrics.values():
```

Comment:
The same logic should apply for LinkableMetrics, too. You can access the outer join path via `LinkableMetric.join_path.semantic_model_join_path`.